next/src/validation/util.ts

```diff
@@ -0,0 +1,49 @@
+export function deepEqual(a: any, b: any): boolean {
+  if (typeof a !== typeof b) {
+    return false
+  }
+
+  if (a === b) {
+    return true
+  }
+
+  // If both are null, the check above has returned true.
+  // If one is null, we return false because we know they are not equal
+  // and since `typeof null === 'object'`, we must not let the null value
+  // pass through as an object.
+  if (a === null || b === null) {
+    return false
+  }
```

Comment:
This is the only change I made to this function while moving it to a dedicated module. I didn't run into the edge case handled here with the json schema test suite tests for `const` but luckily with the `enum` tests.